components/image_viewer.py

```diff
@@ -15,8 +16,59 @@ def layout():
     return html.Div(
         style=COMPONENT_STYLE,
         children=[
-            dcc.Graph(id="image-viewer", config={"scrollZoom": True}),
+            dmc.LoadingOverlay(
+                overlayOpacity=0.35,
```

Comment:
With overlay hidden it would, IMO, be difficult to notice something is happening. I'd prefer to change color of it to make it more obious (right now it's gray), keep it at .15-.35, which would still give a good opacity and udnerstanding to the user that their action just changed the image